### ISSUE TITLE
Popup closes immediately after clicking on subscribe using the factory button - sandbox account

### ISSUE THREAD

**Issue by kvskoker** [paypal/paypal-checkout-components#2026]
The sandbox PayPal subscription factory button was working well before 27/10/2022.

Now the popup closes immediately after clicking on subscribe using the factory button. I get the following errors in the console:

1. POST https://www.sandbox.paypal.com/smart/api/billagmt/subscriptions/I-8PFX6T3MUARY/cartid   [HTTP/2 400 Bad Request 433ms]
2. create_order_error        buttons:1478:170099
3. Uncaught Error: TOKEN_FAILURE         js:2:72960

**Comment by cdmavani** [comment]
I am getting same error with sandbox subscription.

**Comment by joelmartins** [comment]
I am getting same error with sandbox subscription.

**Comment by gawsoftpl** [comment]
I am getting same error with sandbox subscription

**Comment by nunonam** [comment]
I am getting same error with sandbox subscription

**Comment by duhamaher** [comment]
I am getting same error with sandbox subscription

**Comment by Subalee** [comment]
same here

**Comment by manoj007live** [comment]
I am also getting error :-Things don't appear to be working at the moment. Please try again later.
in browser inspect it gives json:-{
    "ack": "contingency",
    "contingency": "TOKEN_FAILURE",
    "meta": {
        "calc": "4efa165435d43",
        "rlog": "rZJvnqaaQhLn%2FnmWT8cSUo8QXTeqJkzg7PvyMHtc7jDKOWrRoXSCL9sewG1iZt%2BToEDpu8RJDV95Kew8GQmgVg_18429587ba7"
    },
    "server": "J1pzppBACy8BAhiufZgZkCgckA0TngFDC37bWFYnvShuhsRvWaA4G4NeJlaujLfw_n78L8vKVOZf4UyHs0PIQ0F3wR6DaarJqzTUlDjSvUfiG6Y1XHpcCtBuCsUly7q_9KGsooDLsdgTMUdyfGEsOKe5XsuD8iM0RZMQocEffD9Xl9eQ6AKo1zLJSarstDpR9TTy9uqU7ePu2tJWNPyJXK8EigGIrWUcfhtkgfeziQWYiu-i6EswesJTBijtMgH-ySzcCgjRwjLgsIUQ0zT8cpUd6qb7Lb0MPOrx5VQjW6W"
}

**Comment by koteisaev** [comment]
Same error, getting this when redirect user to `approve` link from subscription object `links` set,
So this is not issue of PayPal Checkout Component, but some PayPal servers fatal issue/misconfiguration.

**Comment by Sotatek-KienTran** [comment]
same here, paypal please check.

**Comment by koteisaev** [comment]
I opened a support case at www.paypal-support.com and hope for response, but still no one. Probably they literally overflown with such requests now, if these issues are so widespread and happening whether raw API & redirects used or via checkout frontend components

**Comment by hassanabdullah40** [comment]
@koteisaev any response from support?

**Comment by hfd666** [comment]
same question

**Comment by halikhan** [comment]
same issue!

**Comment by matiasstr** [comment]
same problem

**Comment by gawsoftpl** [comment]
Convince client t o change to Stripe?

> > @koteisaev any response from support?
> 
> No response so far. Don't sure what would be easier - get this problem fixed or convince my client to ditch PayPal and keep using something else, Stripe. PayPal is very buggy, but somehow still there are customers who still use it in 2022. And when I talking not about current issue. i spent almost a week already on rebuilding whole checkout experience from scratch to make sure it does not require any client side scripts due to PayPal issue with `counter.cgi` (sic!) being blocked by default privacy settings when PayPal checkout components used. And now this error.

Convince client to change to Stripe? Many clients still want to pay by service via Paypal, because they know this company and want to pay by email. The second argument is that many users have connected many credit cards and bank accounts with Paypal, and prefer to enter email, password without enter credit card. 

Yes, stripe is better for developer, Paypal has no modern api. The question is whether such a large company is able to make a better and more stable API and Docs. Second question is whether good developers want to work in Paypal?

**Comment by kvskoker** [comment]
@gawsoftpl 
Also, PayPal is widely used in many countries while Stripe is currently restricted to only 47 countries. Developers in most countries do not have the option yet to use Stripe for Payment.

Anyways, the problem seams to be resolved.

Please confirm.

**Comment by koteisaev** [comment]
@gawsoftpl 
Here a bit more context of project, and a bit of extra clarity.
We use Stripe from start, first via Zoho, than directly.
My client is in United States, so this Stripe availability for _merchants_ from 47 counties is not a factor here.
Both PayPal and Stripe available for buyers from plenty of countries.
Stripe has great API, which got a bit too complicated after SCA has been introduced, and I has to switch from Charges API to Payment Intents, which was complicated with requirement to stay on site (no redirects).
At some point some our customers started to ask for PayPal as payment method.
Currently used integration was implemented using the PayPal Checkout Component (a wrapper over PayPal Smart Buttons?).
I working on unifying users experience, and simplest way was to migrate to using always redirects, both for Stripe and PayPal, via Stripe Checkout for Stripe, and via PayPal Checkout sessions & Subscriptions  Approval links with redirects for PayPal.
In this context what most puzzles me are these customers who asked for PayPal payment method,
Even from visual point of view PayPal Checkout & Subscription flow pages look literally unpleasant compared to clean Stripe UI.
Since some of these users are US-based, I guess they are just people who used paypal for a while even before Stripe existing, and just do not want type card details visually in a website, compared to PayPal redirect.
I do not have other explanation for asking PayPal.

**Comment by koteisaev** [comment]
> @gawsoftpl Also, PayPal is widely used in many countries while Stripe is currently restricted to only 47 countries. Developers in most countries do not have the option yet to use Stripe for Payment.
> 
> Anyways, the problem seams to be resolved.
> 
> Please confirm.
> 
> Edit: The problem is not totally resolved yet. The popup does not close immediately as previous but after clicking on the subscribe button it shows the error below: **"Things don't appear to be working at the moment. Please try again later."**
> 
> Maybe they will fix it soon.

I do not hope they will fix it ever. That generic error message you see is still the same `TOKEN_FAILURE` "contingency" error they can not fix for like 24 hours.

**Comment by nazarik** [comment]
It's working now

**Comment by koteisaev** [comment]
yes, confirming that.

**Comment by gregjopa** [comment]
Hi everyone, this token failure issue with the PayPal Subscriptions Button in the PayPal sandbox environment is resolved. Sorry for the inconvenience. I'm going to close this out. Feel free to open a new issue if you are still experiencing problems.

**Comment by koteisaev** [comment]
Is there information available about issue cause and what done to prevent it from happening over and over?
